### PR TITLE
GODRIVER-3149: OIDC Spec Cleanup

### PR DESCRIPTION
**Pull request by pmeredit**
GODRIVER-3149

## Summary

This cleans up a few things that changed in the spec during implementation and adds the 3 tests that were missing from previous tickets

## Background & Motivation

OIDC epic

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -2312,13 +2312,14 @@ task_groups:
...
         params:
-          shell: bash
+          binary: bash
           include_expansions_in_env: ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_SESSION_TOKEN"]
-          script: |
-            ${PREPARE_SHELL}
-            ${DRIVERS_TOOLS}/.evergreen/auth_oidc/setup.sh
+          env:
+            MONGODB_VERSION: "8.0"
+          args:
```
Comment (pmeredit) [inline .evergreen/config.yml:2321]: Note that human18 fails if we don't run with MONGODB_VERSION 8.0

Comment (blink1073) [reply .evergreen/config.yml:2321]: That is tracked by https://jira.mongodb.org/browse/PYTHON-4447, which I updated to reflect today's status.

File: cmd/testoidcauth/main.go [hunk 2]
```diff
@@ -38,7 +39,7 @@ func tokenFile(user string) string {
 	return path.Join(oidcTokenDir, user)
 }
```
Comment (pmeredit) [inline cmd/testoidcauth/main.go:41]: The linter doesn't catch misspelled function names, hah!

**Review by matthewdale — APPROVED** [review]
Looks good! 👍

**Pull request merged**